code/model/modelread.cpp

```diff
@@ -462,6 +462,56 @@ void get_user_prop_value(char *buf, char *value)
 	*p1 = c;
 }
 
+// routine to parse out a vec3d from a user property field of an object
+bool get_user_vec3d_value(char *buf, vec3d *value, bool require_brackets)
+{
+	float f1, f2, f3;
+	char closing_bracket = '\0';
+	bool success = false;
+
+	pause_parse();
+	Mp = buf;
+
+	do {
+		// skip white space and equal sign
+		while (isspace(*Mp) || (*Mp == '='))
+			Mp++;
+
+		if (require_brackets)
+		{
+			// look for vector bracket
+			if (*Mp == '{')
+				closing_bracket = '}';
+			else if (*Mp == '[')
+				closing_bracket = ']';
+			else
+				break;
+		}
+
+		// get comma-separated floats
+		if (stuff_float_optional(&f1) != 2)
+			break;
+		if (stuff_float_optional(&f2) != 2)
+			break;
+		if (stuff_float_optional(&f3) != 2)
+			break;
+
+		if (require_brackets)
+		{
+			ignore_white_space();
+
+			if (*Mp != closing_bracket)
+				break;
+		}
+
+		value->xyz = { f1, f2, f3 };
+		success = true;
+	} while (false);
```

Comment:
Why the do-while?

Reply:
It's so that I could use `break`.  I initially just used a bunch of `return`s, but the problem is that you always need to call `unpause_parse`.

Reply:
Ohh, I see. Yes, that is a good reason to do this. Or use a bunch of `goto`s :wink: